libs/providers/go-feature-flag-web/src/lib/go-feature-flag-web-provider.ts

```diff
@@ -0,0 +1,269 @@
+      this._logger?.info(`Change in your configuration flag`);
+      const t: GOFeatureFlagWebsocketResponse = JSON.parse(data);
+      const flagsChanged = this.extractFlagNamesFromWebsocket(t);
+      await this.fetchAll(OpenFeature.getContext(), flagsChanged);
+    }
+    this._websocket.onclose = async () => {
+      this._logger?.info(`Websocket closed, trying to reconnect`);
+      await this.reconnectWebsocket();
+    };
+    this._websocket.onerror = async (event: Event) => {
+      this._logger?.error(`Error while connecting the websocket: ${event}`);
+      await this.reconnectWebsocket();
+    };
+  }
+
+  /**
+   * waitWebsocketFinalStatus is waiting synchronously for the websocket to be in a stable
+   * state (CLOSED or OPEN).
+   * @param socket - the websocket you are waiting for
+   */
+  waitWebsocketFinalStatus(socket: WebSocket): Promise<void> {
+    return new Promise((resolve) => {
+      const checkConnection = () => {
+        if (socket.readyState === WebSocket.OPEN || socket.readyState === WebSocket.CLOSED) {
+          return resolve();
+        }
+        // Wait 5 milliseconds before checking again
+        setTimeout(checkConnection, 5);
+      };
+      checkConnection();
+    });
+  }
+
+
+  /**
+   * extract flag names from the websocket answer
+   */
+  private extractFlagNamesFromWebsocket(wsResp: GOFeatureFlagWebsocketResponse): string[] {
+    let flags: string[] = [];
+    if (wsResp.deleted) {
+      flags = [...flags, ...Object.keys(wsResp.deleted)];
+    }
+    if (wsResp.updated) {
+      flags = [...flags, ...Object.keys(wsResp.updated)];
+    }
+    if (wsResp.added) {
+      flags = [...flags, ...Object.keys(wsResp.added)];
+    }
+    return flags;
+  }
+
+  /**
+   * reconnectWebsocket is using an exponential backoff pattern to try to restart the connection
+   * to the websocket.
+   */
+  async reconnectWebsocket() {
+    let delay = this._websocketRetryInitialDelay;
+    let attempt = 0;
+    while (attempt < this._websocketMaxRetries) {
+      attempt++;
+      await this.connectWebsocket()
+      if (this._websocket !== undefined && this._websocket.readyState === WebSocket.OPEN) {
+        return
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, delay));
+      delay *= this._websocketRetryDelayMultiplier;
+      this._logger?.info(`error while reconnecting the websocket, next try in ${delay} ms (${attempt}/${this._websocketMaxRetries}).`)
+    }
+    this.events.emit(ProviderEvents.Stale, {
+      message: 'impossible to get status from GO Feature Flag (websocket connection stoppe'
+    });
+  }
+
+  onClose(): Promise<void> {
+    this._websocket?.close(1000, "Closing GO Feature Flag provider");
+    return Promise.resolve(undefined);
+  }
+
+  async onContextChange(_: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    this._logger?.debug(`new context provided: ${newContext}`);
+    await this.fetchAll(newContext);
+  }
```

Comment:
```suggestion
  onContextChange(_: EvaluationContext, newContext: EvaluationContext): Promise<void> {
    this._logger?.debug(`new context provided: ${newContext}`);
    return this.fetchAll(newContext);
  }
```

I think you can just return this since it's returning a void Promise.